src/modules/resource-hwloc/resource.c

```diff
@@ -198,7 +198,7 @@ static char *escape_and_join_kvs_path (const char *base,
         suffix = escape_kvs_key (suffix);
 
         if (!ret_str || !strlen (ret_str))
-            ret_str = suffix;
+            ret_str = strdup (suffix);
```

Comment:
should that be an `xstrdup(`) to match the `xasprintf()` on the other leg of the conditional?